elide-core/src/main/antlr4/com/yahoo/elide/generated/parsers/Expression.g4

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016 Yahoo! Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+
+grammar Expression;
+
+start: expression;
+
+/*
+Doesn't work with single character names.
+*/
+
+expression
+    : NOT WS expression                             #NOT
+    | LPAREN WS? expression WS? RPAREN              #PAREN
+    | left=expression WS AND WS right=expression    #AND
+    | left=expression WS  OR WS right=expression    #OR
+    | permissionClass                               #PERMISSION
+    ;
+
+/*
+No only numeric because class names can't be only numeric.
+Do not name this 'class' as that is a keyword.
+*/
+permissionClass: LEGAL_NAME;
+
+NOT         : 'NOT' | 'Not' | 'not';
```

Comment:
I think it's important to maintain consistency. nOt may not be conventional, but it is allowed in SQL and other language whose operators are case insensitive. Either we make it case sensitive and only allow one form, or we make it case insensitive and allow all forms.